src/Nodes/Endpoints/EndpointsLabelNode.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import IotHubClient from "azure-arm-iothub";
+import * as vscode from "vscode";
+import { Constants } from "../../constants";
+import { TelemetryClient } from "../../telemetryClient";
+import { Utility } from "../../utility";
+import { CommandNode } from "../CommandNode";
+import { INode } from "../INode";
+import { BuiltInEndpointLabelNode } from "./BuiltInEndpointLabelNode";
+import { CustomEndpointLabelNode } from "./CustomEndpointLabelNode";
+import { EventHubLabelNode } from "./EventHubLabelNode";
+
+export class EndpointsLabelNode implements INode {
+    constructor() {
+    }
+
+    public getTreeItem(): vscode.TreeItem {
+        return {
+            label: "Endpoints",
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            contextValue: "endpoints-label",
+        };
+    }
+
+    public async getChildren(): Promise<INode[]> {
+        TelemetryClient.sendEvent(Constants.IoTHubAILoadEndpointsTreeStartEvent);
+
+        try {
+            const accountApi = Utility.getAzureAccountApi();
+            const subscriptionId = Constants.ExtensionContext.globalState.get(Constants.StateKeySubsID);
+            if (!subscriptionId || !(await accountApi.waitForLogin())) {
+                return [new CommandNode("-> Please select an IoT Hub", "azure-iot-toolkit.selectIoTHub")];
+            }
+
+            const subscription = accountApi.subscriptions.find((element) => element.subscription.subscriptionId === subscriptionId);
+            const client = new IotHubClient(subscription.session.credentials, subscription.subscription.subscriptionId, subscription.session.environment.resourceManagerEndpointUrl);
+            const iotHubs = await client.iotHubResource.listBySubscription();
+            const iothub = iotHubs.find((element) =>
+                element.id === Constants.ExtensionContext.globalState.get(Constants.StateKeyIoTHubID));
+            TelemetryClient.sendEvent(Constants.IoTHubAILoadEndpointsTreeDoneEvent, { Result: "Success" });
+
+            return [new BuiltInEndpointLabelNode(),
+                new EventHubLabelNode(subscription, iothub.properties.routing.endpoints.eventHubs),

```

Comment:
shall we check iothub before using it? It seems it is possible no iothub is got?

Reply:
Good catch!